File: rust/lib/src/execution/test_suite_configurator.rs
use anyhow::Result;

use crate::testsuite::testsuite::TestSuite;

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
pub trait TestSuiteConfigurator {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	fn set_log_level(&self, log_level_str: &str) -> Result<()>;

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	fn parse_params_and_create_suite(&self, params_json_str: &str) -> Result<Box<dyn TestSuite>>;
}

File: golang/lib/execution/test_suite_configurator.go
/*
 * Copyright (c) 2021 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package execution

import "github.com/kurtosis-tech/kurtosis-libs/golang/lib/testsuite"

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
type TestSuiteConfigurator interface {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	SetLogLevel(logLevelStr string) error

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	ParseParamsAndCreateSuite(paramsJsonStr string) (testsuite.TestSuite, error)
}
